### PR TITLE
enhancement: simplify ready and health check handler usage

### PR DESCRIPTION
**Pull request by fschade**
## Description
WIP

## Related Issue
-  https://github.com/owncloud/ocis/issues/10281

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: ocis-pkg/handlers/checker.go [hunk 1]
```diff
@@ -43,19 +45,21 @@ func (c CheckHandlerConfiguration) WithLogger(l log.Logger) CheckHandlerConfigur
...
+
 	return c
 }
 
-// WithInheritedChecksFrom appends the checks from another CheckHandlerConfiguration.
-func (c CheckHandlerConfiguration) WithInheritedChecksFrom(other CheckHandlerConfiguration) CheckHandlerConfiguration {
-	for name, check := range other.Checks {
-		c.WithCheck(name, check)
+// WithChecks adds multiple checks to the CheckHandlerConfiguration.
+func (c CheckHandlerConfiguration) WithChecks(checks checkers) CheckHandlerConfiguration {
```
Comment (dragonchaser) [inline ocis-pkg/handlers/checker.go:60]: This is easily confused with `WithChecks` at least rename it to `WithChecksFrom`

**Review by dragonchaser — CHANGES_REQUESTED** [review]

File: ocis-pkg/checks/checkgrpc.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 )
 
 // NewGRPCCheck checks the reachability of a grpc server.
-func NewGRPCCheck(address string) func(ctx context.Context) error {
+func NewGRPCCheck(address string) func(context.Context) error {
```
Comment (dragonchaser) [inline ocis-pkg/checks/checkgrpc.go:13]: I thought we agreed on dropping the context for now, because it has no use atm.

**Review by dragonchaser — CHANGES_REQUESTED** [review]

File: ocis-pkg/checks/checkgrpc.go (hunk 2 above)
Comment (fschade) [reply ocis-pkg/checks/checkgrpc.go:13]: It hast, its for cancellation of the err group, see the tests

File: ocis-pkg/handlers/checker.go (hunk 1 above)
Comment (fschade) [reply ocis-pkg/handlers/checker.go:60]: I’m not a big fan at all the check handler can inherit from other handlers, I still think all those things should happen in the configuration only.

Is we name it ‘from’ it indicates that it uses the checks from xyz and not the checks directly, but if you want it that way, can you please push it? I’m not at my desk at the moment

**Review by dragonchaser — APPROVED** [review]

**Pull request merged**